app/src/main/java/com/example/dinhh/soundscape/device/AddSound.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.dinhh.soundscape.device
+
+import android.media.MediaPlayer
+import com.example.dinhh.soundscape.data.Model
+import com.example.dinhh.soundscape.data.entity.SoundscapeItem
+import io.reactivex.Completable
+import java.io.IOException
+
+interface AddSound {
+    fun addSound(selectedSound: String, selectedPosition: Int, title: String, length: Int, category: String, volume: Int): Completable
```

Comment:
Device layer should not know anything about UI layer. Prefererbly not pass selectedSound selectedPosition ...

Reply:
selectedSound is the download url for the sound so it needs to be there, but that selectedPosition parameter was a part of the functionality that changed the stop button back to play button after the sound file had ended, but it doesn't work anymore, so I can get rid of that one.

Reply:
I will try to refactor it and you can take a look on that

Reply:
You are right. It turns out that selectedSound is soundUrl (download link). The name made me misunderstood so I change it to soundUrl